src/Microsoft.ML.Data/Transforms/ValueMapping.cs

```diff
@@ -73,9 +73,13 @@ public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
                 if (!inputSchema.TryFindColumn(inputColumnName, out var originalColumn))
                     throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", inputColumnName);
 
-                if ((originalColumn.Kind == SchemaShape.Column.VectorKind.VariableVector ||
-                   originalColumn.Kind == SchemaShape.Column.VectorKind.Vector) && Transformer.ValueColumnType is VectorType)
-                    throw Host.ExceptNotSupp("Column '{0}' cannot be mapped to values when the column and the map values are both vector type.", inputColumnName);
+                if (originalColumn.Kind == SchemaShape.Column.VectorKind.VariableVector ||
+                   originalColumn.Kind == SchemaShape.Column.VectorKind.Vector)
```

Comment:
it seems the bug is that we have not updated the `vectorKind`  when the `originalColumn.Kind == SchemaShape.Column.VectorKind.VariableVector`

As such , instead of changing this logic should we just update the vectorKind ? i.e. add the lines:

 if (originalColumn.Kind == SchemaShape.Column.VectorKind.VariableVector)
                    vectorKind = SchemaShape.Column.VectorKind.VariableVector;

Reply:
I think this what's happening right now both for fixed length and variable length vector.

---
In reply to: [253654474](https://github.com/dotnet/machinelearning/pull/2386#discussion_r253654474) [](ancestors = 253654474)